src/GoTrueClient.ts

```diff
@@ -216,78 +219,82 @@ export default class GoTrueClient {
 
   /**
    * Log in an existing user, or login via a third-party provider.
-   * @type UserCredentials
+   * @type SignInWithPasswordCredentials
    * @param email The user's email address.
    * @param phone The user's phone number.
    * @param password The user's password.
-   * @param refreshToken A valid refresh token that was returned on login.
-   * @param provider One of the providers supported by GoTrue.
-   * @param redirectTo A URL to send the user to after they are confirmed (OAuth logins only).
-   * @param shouldCreateUser A boolean flag to indicate whether to automatically create a user on magiclink / otp sign-ins if the user doesn't exist. Defaults to true.
-   * @param scopes A space-separated list of scopes granted to the OAuth application.
+   * @param options Valid options for password sign-ins.
    */
-  async signIn(
-    { email, phone, password, refreshToken, provider, oidc }: UserCredentials,
-    options: {
-      redirectTo?: string
-      shouldCreateUser?: boolean
-      scopes?: string
-      captchaToken?: string
-      queryParams?: { [key: string]: string }
-    } = {}
-  ): Promise<
+  async signInWithPassword({
+    email,
+    phone,
+    password,
+    options = {},
+  }: SignInWithPasswordCredentials): Promise<
     | {
         session: Session | null
         user: User | null
-        provider?: Provider
         url?: string | null
         error: null
       }
     | {
         session: Session | null
         user: User | null
-        provider?: Provider
         url?: string | null
         error: AuthError
       }
   > {
     try {
       this._removeSession()
 
-      if (email && !password) {
-        const { error } = await this.api.sendMagicLinkEmail(email, {
-          redirectTo: options.redirectTo,
-          shouldCreateUser: options.shouldCreateUser,
-          captchaToken: options.captchaToken,
-        })
-        return { user: null, session: null, error }
-      }
       if (email && password) {
         return this._handleEmailSignIn(email, password, {
-          redirectTo: options.redirectTo,
-        })
-      }
-      if (phone && !password) {
-        const { error } = await this.api.sendMobileOTP(phone, {
-          shouldCreateUser: options.shouldCreateUser,
           captchaToken: options.captchaToken,
         })
-        return { user: null, session: null, error }
       }
       if (phone && password) {
         return this._handlePhoneSignIn(phone, password)
       }
-      if (refreshToken) {
-        // currentSession and currentUser will be updated to latest on _callRefreshToken using the passed refreshToken
-        const { error } = await this._callRefreshToken(refreshToken)
-        if (error) throw error
-
-        return {
-          user: this.currentUser,
-          session: this.currentSession,
-          error: null,
-        }
+      throw new Error(`You must provide either an email or phone number and a password.`)
```

Comment:
Since this is for v2 it would be good to have proper error codes and responses. And to keep these consistent between libraries along with a glossy explaining what each error means and what could be the cause.

Reply:
resolved in https://github.com/supabase/gotrue-js/pull/341